src/tuner/nccl_ofi_tuner.c

```diff
@@ -41,327 +41,351 @@ ncclResult_t nccl_ofi_tuner_destroy(void *context)
-				{.algorithm = NCCL_ALGO_RING,
-				 .protocol = NCCL_PROTO_LL128,
-				 .num_vertices = 9,
-				 .vertices = {{0, 2},
-					      {4736000, 2},
-					      {269484032, 128},
-					      extended_ring_ll128,
-					      extended_tree_ll128,
-					      {119477248, 128},
-					      {9999360, 64},
-					      {2367488, 16},
-					      {0, 16}}},
-				{.algorithm = NCCL_ALGO_RING,
-				 .protocol = NCCL_PROTO_SIMPLE,
-				 .num_vertices = 4,
-				 .vertices = {{4736000, 2}, {TUNER_MAX_SIZE, 2}, extended_ring_ll128, {269484032, 128}}}};
-
-			ret = set_regions(nccl_ofi_tuner_ctx, collType, sizeof(regions) / sizeof(regions[0]), regions);
-			if (ret != ncclSuccess) {
-				goto exit;
-			}
-		}
-		{
-			collType = ncclFuncAllGather;
-			nccl_ofi_tuner_point_t extended_ring_simple =
-				extend_region((nccl_ofi_tuner_point_t){4194304, 2},
-					      (nccl_ofi_tuner_point_t){8589934592, 2048},
-					      (nccl_ofi_tuner_point_t){TUNER_MAX_SIZE, TUNER_MAX_RANKS});
-
-			const nccl_ofi_tuner_region_t regions[] = {{.algorithm = NCCL_ALGO_RING,
-								    .protocol = NCCL_PROTO_SIMPLE,
-								    .num_vertices = 4,
-								    .vertices = {{4194304, 2},
-										 {TUNER_MAX_SIZE, 2},
-										 extended_ring_simple,
-										 {8589934592, 2048}}}};
-
-			ret = set_regions(nccl_ofi_tuner_ctx, collType, sizeof(regions) / sizeof(regions[0]), regions);
-			if (ret != ncclSuccess) {
-				goto exit;
-			}
-		}
-		{
-			collType = ncclFuncReduceScatter;
-			nccl_ofi_tuner_point_t extended_ring_simple =
-				extend_region((nccl_ofi_tuner_point_t){8388608, 2},
-					      (nccl_ofi_tuner_point_t){4294967296, 1024},
-					      (nccl_ofi_tuner_point_t){TUNER_MAX_SIZE, TUNER_MAX_RANKS});
-
-			const nccl_ofi_tuner_region_t regions[] = {{.algorithm = NCCL_ALGO_RING,
-								    .protocol = NCCL_PROTO_SIMPLE,
-								    .num_vertices = 4,
-								    .vertices = {{8388608, 2},
-										 {TUNER_MAX_SIZE, 2},
-										 extended_ring_simple,
-										 {4294967296, 1024}}}};
-
-			ret = set_regions(nccl_ofi_tuner_ctx, collType, sizeof(regions) / sizeof(regions[0]), regions);
-			if (ret != ncclSuccess) {
-				goto exit;
-			}
-		}
-	} else {
-		/* Fall back to NCCL's tuner, so no regions */
-	}
-
-	NCCL_OFI_INFO(NCCL_INIT | NCCL_TUNING, "Tuner init: comm with %ld ranks and %ld nodes.", nRanks, nNodes);
+  ncclResult_t ret = ncclSuccess;
+  *context = NULL;
+
+  ofi_log_function = logFunction;
+  ncclFunc_t collType;
+
+  nccl_net_ofi_mutex_lock(&nccl_ofi_tuner_ctx_lock);
+  nccl_ofi_tuner_context_t *nccl_ofi_tuner_ctx =
+      (nccl_ofi_tuner_context_t *)calloc(1, sizeof(nccl_ofi_tuner_context_t));
+  if (nccl_ofi_tuner_ctx == NULL) {
+    NCCL_OFI_WARN("Context allocation failed.");
+    ret = ncclInternalError;
+    goto exit;
+  }
+  nccl_ofi_tuner_ctx->dims.num_ranks = nRanks;
+  nccl_ofi_tuner_ctx->dims.num_nodes = nNodes;
+
+  struct ec2_platform_data *platform_data = NULL;
```

Comment:
All you're using is the platform name.  Don't do all the work exposing the platform data when all you need is to expose `get_platform_type()`.  Of course, fix the namespacing of get_platform_type first, since it will no longer be a static function.

Reply:
Done. Moved just the get_platform_type() into the header.